### PR TITLE
Define proper user type for global resource

### LINKED ISSUE #3956: Creating a new resource at the 'Global' level sets the user group as 'prefs'
Probably related to #3918, #3828, and #3780

When creating a new app resource at the 'Global' level, it is currently added to the User Group `Prefs` instead of `Common` as it was before!

### Correct Behavior:
`v7.8.0`:
image

### Incorrect Behavior:
`v7.8.13`:
image

This should be resolved and investigated ASAP.

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #3956

### CODE REVIEW

**Comment by CarolineDenis** [comment]
NOTES: 
- don't forget to uncomment spAppResourceDir in webOnlyViews 
- add a way to only choose from Preferences or Global Preferences when "Other Properties Resource" is chosen?

**Comment by CarolineDenis** [comment]
@maxpatiiuk , 
Can we create two new keys one for Preferences and one for Global Preferences that would have a name? We would then use those to create them instead of otherPropertiesResource. 
https://github.com/specify/specify7/blob/84e56043528e3fa4c765d31da0f186882de86116/specifyweb/frontend/js_src/lib/components/AppResources/types.tsx#L199

File: specifyweb/frontend/js_src/lib/components/AppResources/tree.ts [hunk 1]
```diff
@@ -67,13 +67,25 @@ function getGlobalAllResources(resources: AppResources): {
...
-  if (globalDirectories.length === 0)
-    globalDirectories.push({
-      ...addMissingFields('SpAppResourceDir', {
-        userType: 'Common',
-      }),
-      scope: 'global',
-    });
+
+  const directoriesNeeded = ['Global Prefs', 'Prefs', 'Common'];
+  if (globalDirectories.length < directoriesNeeded.length) {
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/AppResources/tree.ts:72]: this condition should be removed - this assumes that there are only 3 possible user types at a global level - not an assumption I am willing to make given that these directories are created by sp6 - who knows what it does

File: specifyweb/frontend/js_src/lib/components/AppResources/tree.ts [hunk 2]
```diff
@@ -67,13 +67,25 @@ function getGlobalAllResources(resources: AppResources): {
...
+    const directoriesPresent = new Set(
+      globalDirectories.map((directory) => directory.userType)
+    );
+    const missingUserType = directoriesNeeded.find(
+      (userType) => !directoriesPresent.has(userType)
+    );
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/AppResources/tree.ts:78]: this code assumes that at most 1 user type is missing - can you rewrite to add any user type that is missing, not just one?

File: specifyweb/frontend/js_src/lib/components/AppResources/types.tsx [hunk 3]
```diff
@@ -203,6 +203,20 @@ export const appResourceSubTypes = ensure<IR<AppResourceSubType>>()({
...
+    icon: icons.viewList,
+    label: preferencesText.preferences(),
+  },
+  globalPreferences: {
+    mimeType: 'text/x-java-properties',
+    name: 'Global Preferences',
+    documentationUrl: undefined,
+    icon: icons.viewList,
+    label: resourcesText.globalPreferences(),
+  },
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/AppResources/types.tsx:219]: we are not encouraging users to create these app resources (they should be created by sp6 in every database). In fact, we should deprecate these app resources as soon as Institutional Preferences are added), thus we should not be exposing explicit options for these legacy resources in the UI.

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

**Comment by maxpatiiuk** [comment]
see also https://github.com/specify/specify7/issues/3956#issuecomment-1853251752 for proposed solutions

**Pull request closed without merging**